### PR TITLE
SendCode router update, Smtp client setup.

### PR DESCRIPTION
**Pull request by nyzd**
From now sendcode router will send verfiy code to the email.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -11,4 +11,5 @@ serde = { version = "1.0", features = ["derive"] }
 serde_json = "1"
 rand = "0.8.5"
 chrono = "0.4"
-sha2 = "0.10.6"
\ No newline at end of file
+sha2 = "0.10.6"
+lettre = { version = "0.10.1", default-features = false, features = ["smtp-transport", "tokio1-rustls-tls", "hostname", "builder"] }
```
Comment (nyzd) [inline Cargo.toml:15]: Crate to send email (smtp client )

File: src/email.rs [hunk 2]
```diff
@@ -0,0 +1,49 @@
+use lettre::{
+    transport::smtp::authentication::Credentials, AsyncSmtpTransport, AsyncTransport, Message,
+    Tokio1Executor,
+};
+use std::error::Error;
+
+#[derive(Clone)]
+pub struct EmailManager {
+    default_from: String,
```
Comment (nyzd) [inline src/email.rs:9]: For example. `natiq <info@natiq.net>`

File: src/email.rs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+impl EmailManager {
+    /// Grab creds and create smtpTransport and return EmailManage
+    pub fn new(
+        host: &str,
+        port: u16,
+        creds: Credentials,
+        default_from: String,
+    ) -> Result<Self, Box<dyn Error>> {
```
Comment (nyzd) [inline src/email.rs:20]: The box is a heap allocation when the compiler doesn't know the size of the data. Error is a trait and traits don't have size.

File: src/main.rs [hunk 4]
```diff
@@ -30,9 +48,12 @@ async fn main() -> std::io::Result<()> {
         .build(pg_manager)
         .expect("Failed to create pool.");
 
+    let mailer = create_emailer();
+
     HttpServer::new(move || {
         App::new()
             .app_data(web::Data::new(pool.clone()))
+            .app_data(web::Data::new(mailer.clone()))
```
Comment (nyzd) [inline src/main.rs:56]: Is it a good solution?

**Review by nyzd — COMMENTED** [review]
Some comments

**Review by al-abd — APPROVED** [review]
Please write comments inside code

**Pull request merged**